games-rpg/soltys/soltys-1.0-r2.ebuild

```diff
@@ -1,64 +1,52 @@
-# Copyright 1999-2019 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-inherit eutils
+EAPI=8
+
+inherit desktop wrapper
 
 DESCRIPTION="Classic adventure game"
 HOMEPAGE="https://wiki.scummvm.org/index.php/Soltys"
 SRC_URI="l10n_en? ( mirror://sourceforge/scummvm/${PN}-en-v${PV}.zip )
 	l10n_es? ( mirror://sourceforge/scummvm/${PN}-es-v${PV}.zip )
 	l10n_pl? ( mirror://sourceforge/scummvm/${PN}-pl-v${PV}.zip )
 	!l10n_en? ( !l10n_es? ( !l10n_pl? ( mirror://sourceforge/scummvm/${PN}-en-v${PV}.zip ) ) )
-	http://www.scummvm.org/images/cat-soltys.png"
+	http://www.scummvm.org/images/cat-${PN}.png"
+S="${WORKDIR}"
 
 LICENSE="Soltys"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 IUSE="l10n_en l10n_es l10n_pl"
 
-RDEPEND=">=games-engines/scummvm-1.5"
-DEPEND="app-arch/unzip"
-
-S="${WORKDIR}"
+RDEPEND="games-engines/scummvm"
+BDEPEND="app-arch/unzip"
 
 src_unpack() {
-	if use l10n_en || ( ! use l10n_en && ! use l10n_es && ! use l10n_pl ) ; then
-		mkdir -p en || die
-		unpack ${PN}-en-v${PV}.zip
-		mv vol.{cat,dat} en/ || die
-	fi
-	if use l10n_es ; then
-		mkdir -p es || die
-		unpack ${PN}-es-v${PV}.zip
-		mv soltys-es-v1-0/vol.{cat,dat} es/ || die
-	fi
-	if use l10n_pl ; then
-		mkdir -p pl || die
-		unpack ${PN}-pl-v${PV}.zip
-		mv vol.{cat,dat} pl/ || die
-	fi
+	MY_L10N=( $(usev l10n_en) $(usev l10n_es) $(usev l10n_pl) )
+	[[ ${MY_L10N} ]] || MY_L10N=( l10n_en )
+
+	local lang
+	for lang in "${MY_L10N[@]//l10n_/}"; do
+		mkdir -p ${lang} || die
+		unpack ${PN}-${lang}-v${PV}.zip
+		mv vol.{cat,dat} ${lang}/ || die
+	done
 }
 
 src_prepare() {
 	default
-	rm -rf license.txt soltys-es-v1-0
+	rm license.txt || die
 }
 
 src_install() {
 	insinto /usr/share/${PN}
 	doins -r *
-	newicon "${DISTDIR}"/cat-soltys.png soltys.png
-	if use l10n_en || ( ! use l10n_en && ! use l10n_es && ! use l10n_pl ) ; then
-		make_wrapper soltys-en "scummvm -f -p \"/usr/share/${PN}/en\" soltys" .
-		make_desktop_entry ${PN}-en "Soltys (English)" soltys
-	fi
-	if use l10n_es ; then
-		make_wrapper soltys-es "scummvm -f -p \"/usr/share/${PN}/es\" soltys" .
-		make_desktop_entry ${PN}-es "Soltys (Español)" soltys
-	fi
-	if use l10n_pl ; then
-		make_wrapper soltys-pl "scummvm -f -p \"/usr/share/${PN}/pl\" soltys" .
-		make_desktop_entry ${PN}-pl "Soltys (Polski)" soltys
-	fi
+	newicon "${DISTDIR}"/cat-${PN}.png ${PN}.png
+
+	local lang
+	for lang in "${MY_L10N[@]//l10n_/}"; do
+		make_wrapper ${PN}-${lang} "scummvm -f -p \"${EPREFIX}/usr/share/${PN}/${lang}\" ${PN}" .
+		make_desktop_entry ${PN}-${lang} "Soltys (${lang})" ${PN}
```

Comment:
The third argument on either of those isn't needed for any of these ebuilds.
- `.` makes the wrapper change directory.. to the current dir, unspecifying does nothing which is current dir
- `${PN}` is the default for icon if unspecified